src/ray/util/process.h

```diff
@@ -0,0 +1,91 @@
+#ifndef RAY_UTIL_PROCESS_H
+#define RAY_UTIL_PROCESS_H
+
+#include <functional>  // std::equal_to, std::hash, std::less
+#include <memory>      // std::shared_ptr
+#include <utility>     // std::forward
+
+#include <boost/process/args.hpp>
+#include <boost/process/child.hpp>
+
+// We only define operators required by the standard library (<, ==, hash).
+// We declare but avoid defining the rest so that they're not used by accident.
+
+namespace std {
+template <>
+struct hash<boost::process::child> : hash<boost::process::pid_t> {
+  typedef hash<boost::process::pid_t> base_type;
+  typedef boost::process::child argument_type;
+  typedef size_t result_type;
+  result_type operator()(const argument_type &value) const {
+    return this->base_type::operator()(value.id());
+  }
+};
+template <>
+struct equal_to<boost::process::child> : equal_to<boost::process::pid_t> {
+  typedef equal_to<boost::process::pid_t> base_type;
+  typedef boost::process::child argument_type;
+  typedef bool result_type;
+  result_type operator()(const argument_type &a, const argument_type &b) const {
+    return this->base_type::operator()(a.id(), b.id());
+  }
+};
+template <>
+struct greater<boost::process::child>;
+template <>
+struct greater_equal<boost::process::child>;
+template <>
+struct less<boost::process::child>;
+template <>
+struct less_equal<boost::process::child>;
+template <>
+struct not_equal_to<boost::process::child>;
+
+// Invalid process objects must be distinguished by their addresses.
```

Comment:
This is very magical, we should get rid of it before merging the PR

Reply:
I'm not sure we can unfortunately